### PR TITLE
[Feature] Add visual placeholder for TextInput

### LINKED ISSUE #259: Implement NONACCESSIBLE-placeholder for TextInput
## The Problem

VRK is using non-accessible placeholder attribute in designs and services.

## Proposed Solution

`placeholder` will keep warning that it is not accessible.
Implement `NONACCESSIBLE-placeholder`-prop

Add to typescript-interface:
```ts
'NONACCESSIBLE-placeholder'?: string;
```

## Additional Info

https://www.smashingmagazine.com/2018/06/placeholder-attribute/
https://www.nngroup.com/articles/form-design-placeholders/

### PR DESCRIPTION
**Pull request by LJKaski**
## Description
TextInput had a somewhat strange placeholder logic since inaccessible use of placeholders wanted to be kept to the minimum. This PR introduces a visualPlaceholder prop that gets passed to the component as placeholder.

These changes also slightly affected SearchInput, so the necessary changes were made to accommodate the changes.

The examples were also adjusted to show the new prop and logic.

## Related Issue
https://github.com/vrk-kpa/suomifi-ui-components/issues/259

Closes #259

## Motivation and Context
Placeholders are in use in many of the designs for Suomi.fi. To aid the implementation of the ui-components, a concession was made to implement it. However, the API description should guide the developers to the right track in terms of placeholders.

## How Has This Been Tested?
Yarn test and running locally

## Release notes
Implemented placeholder text for TextInput

### CODE REVIEW

File: src/components/Form/TextInput.tsx [hunk 1]
```diff
@@ -48,13 +47,12 @@ export interface TextInputProps extends Omit<HtmlInputProps, 'type'> {
   onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
   /** Label */
   labelText: string;
-  /** Label displaymode -
-   * visible: show above, hidden: show as placeholder
+  /** Hide or show label. Label element is always present, but can be visually hidden.
```
[the remark this answers is not in the document]
Comment (LJKaski) [reply src/components/Form/TextInput.tsx:50]: Good points. I think it's better to keep the API description short and put those points in the DS-site documentation.

File: src/core/Form/TextInput/TextInput.md [hunk 2]
```diff
@@ -1,10 +1,17 @@
 ```js
 import { TextInput } from 'suomifi-ui-components';
-
-<TextInput
-  onBlur={(event) => console.log(event.target.value)}
-  labelText="Test TextInput"
-/>;
+<>
```
[the remark this answers is not in the document]
Comment (LJKaski) [reply src/core/Form/TextInput/TextInput.md:3]: That's probably a good idea. The tests didn't say anything about these changes even though the API changed...

[the remark this answers is not in the document]
Comment (LJKaski) [reply src/core/Form/TextInput/TextInput.md:3]: Added a couple more tests to cover more of the API. We can add more thorough tests after we have all the functionalities in place (hopefully soon).

File: src/core/Form/TextInput/TextInput.test.tsx [hunk 3]
```diff
@@ -8,10 +8,43 @@ const TestTextInput = (
...
+  />
+);
+
 test('calling render with the same component on the same container does not remount', () => {
   const buttonRendered = render(TestTextInput);
   const { container } = buttonRendered;
   expect(container.firstChild).toMatchSnapshot();
 });
 
+test('calling render with the same component on the same container does not remount', () => {
```
Comment (ketsappi) [inline src/core/Form/TextInput/TextInput.test.tsx:38]: Would be better to have unique and more describing names for each tests.
Or at least add 1, 2, 3 if it is not easy to come with better names, as now tests look like this when ran:

```
  ✓ calling render with the same component on the same container does not remount (74ms)
  ✓ calling render with the same component on the same container does not remount (34ms)
  ✓ calling render with the same component on the same container does not remount (24ms)
```

Comment (aappoalander) [reply src/core/Form/TextInput/TextInput.test.tsx:38]: I agree, I think it' so quick to fix, that let's do it right away? :)

File: src/components/Form/TextInput.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aappoalander) [reply src/components/Form/TextInput.tsx:50]: Maybe the visually hidden is sufficient in this case.

File: src/core/Form/TextInput/TextInput.test.tsx (hunk 3 above)
Comment (LJKaski) [reply src/core/Form/TextInput/TextInput.test.tsx:38]: For me the tests were automatically numbered. I added more descriptive names now.

Comment (ketsappi) [reply src/core/Form/TextInput/TextInput.test.tsx]: Now the tests names are looking nice 👍

**Pull request merged**